nucypher/policy/conditions/offchain.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Optional, Tuple
+
+import requests
+from jsonpath_ng.exceptions import JsonPathLexerError, JsonPathParserError
+from jsonpath_ng.ext import parse
+from marshmallow import ValidationError, fields, post_load, validate
+from marshmallow.fields import Field
+
+from nucypher.policy.conditions.base import AccessControlCondition
+from nucypher.policy.conditions.exceptions import (
+    ConditionEvaluationFailed,
+    InvalidCondition,
+)
+from nucypher.policy.conditions.lingo import ConditionType, ReturnValueTest
+from nucypher.policy.conditions.utils import CamelCaseSchema
+from nucypher.utilities.logging import Logger
+
+
+class JSONPathField(Field):
+    default_error_messages = {"invalid": "Not a valid JSONPath expression."}
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if not isinstance(value, str):
+            self.fail("invalid")
+        try:
+            parse(value)
+        except (JsonPathLexerError, JsonPathParserError):
+            self.fail("invalid")
+        return value
+
+
+class OffchainCondition(AccessControlCondition):
+    """
+    An offchain condition is a condition that can be evaluated by reading from a JSON
+    endpoint. This may be a REST service but the only requirement is that
+    the response is JSON and can be parsed using jsonpath.
+    """
+
+    CONDITION_TYPE = ConditionType.OFFCHAIN.value
+    LOGGER = Logger("nucypher.policy.conditions.offchain")
+
+    class Schema(CamelCaseSchema):
+
+        name = fields.Str(required=False)
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.OFFCHAIN.value), required=True
+        )
+        headers = fields.Dict(required=False)
+        parameters = fields.Dict(required=False)
+        endpoint = fields.Str(required=True)
+        query = JSONPathField(required=True)
+        return_value_test = fields.Nested(
+            ReturnValueTest.ReturnValueTestSchema(), required=True
+        )
+
+        def validate_query(self, value):
+            try:
+                parse(value)
+            except Exception as e:
+                raise ValidationError(f"Invalid JSONPath query: {e}")
+
+        @post_load
+        def make(self, data, **kwargs):
+            return OffchainCondition(**data)
+
+    def __init__(
+        self,
+        endpoint: str,
+        query: Optional[str],
+        return_value_test: ReturnValueTest,
+        headers: Optional[dict] = None,
```

Comment:
Obviously not related to this PR but rather the ones that build on it, I'm intrigued by the use of headers to specify the "Authentication" header. It's an interesting idea.